### DISCUSSION TITLE
What is this statement used for in remote command parsing rcommand

### DISCUSSION THREAD

**Discussion by ScholliYT** [cyberman54/ESP32-Paxcounter#785] [Q&A] [answered]
Hi,

when reading through the code I  stumbled across this [line](https://github.com/cyberman54/ESP32-Paxcounter/blob/36363ee3771c47a39185c14685fec5af8ce38474/src/rcommand.cpp#L429). Does anyone know what that increment is for? After executing this statement the function is exited.

**Comment by cyberman54** [comment, marked as answer]
This forwards the cursor (a pointer to commands or arguments) of the command parsing loop to the next command in the buffer, if there is one (_while (cursor < cmdlength_). This allows to send multiple commands in one downlink.

**Reply by ScholliYT** [reply]
Ahh yes, now I see why. I missed the outer loop
